Runtime/Triangulator.cs

```diff
@@ -1430,45 +1442,18 @@ public void Execute()
                 using var _pathPoints = pathPoints = new NativeList<int>(Allocator.Temp);
                 using var _pathHalfedges = pathHalfedges = new NativeList<int>(Allocator.Temp);
                 using var _visitedTriangles = visitedTriangles = new NativeList<bool>(triangles.Length / 3, Allocator.Temp);
-                using var _constrainedHalfedges = constrainedHalfedges = new NativeList<bool>(triangles.Length, Allocator.Temp) { Length = triangles.Length };
 
                 using var heQueue = new NativeList<int>(triangles.Length, Allocator.Temp);
                 using var tQueue = new NativeList<int>(triangles.Length, Allocator.Temp);
 
-                if (!constraints.IsCreated || !restoreBoundary)
+                if (constrainBoundary)
```

Comment:
Imo, this shouldn't be done inside the RefineMesh method. Do it before calling this method instead. It affects the constrainedHalfedges list, which is kept outside the method.

```
if (!input.ConstraintEdges.IsCreated || !restoreBoundary) ConstrainBoundary(constrainedHalfedges, halfedges);
```

Reply:
I've reconsidered this and agree with the suggestion. However, there are already circles calculated in the "job" at the beginning of the `Execute` method. To maintain a unified approach, I will keep this for now in the job, but it will be refactored later.